chain/network/src/peer/peer_actor.rs

```diff
@@ -1132,6 +1134,17 @@ impl PeerActor {
             }
             PeerMessage::PeersResponse(PeersResponse { peers, direct_peers }) => {
                 tracing::debug!(target: "network", "Received peers from {}: {} peers and {} direct peers.", self.peer_info, peers.len(), direct_peers.len());
+
+                // Check for abusive behavior (sending too many peers)
+                if peers.len() > PEERS_RESPONSE_MAX_PEERS.try_into().unwrap() {
```

Comment:
Since I'm not familiar with the surrounding context, I have some questions:

- How does the responding side know how not to exceed the limit? Does the request specify a max limit, or does it come from a configuration, or does it come from a statically compiled constant in the code? If it's the latter two, how do we make sure we don't ban a peer because their config/version is simply different?
- Same question for the tier 2 peer count limit below.

Reply:
Thanks for taking a look. `PEERS_RESPONSE_MAX_PEERS` and `MAX_TIER2_PEERS` are both statically compiled constants. It's true that as implemented, modifying those constants in future versions would pose a challenge.

I like the idea of specifying expected limits in the PeersRequest.

Reply:
I've implemented it and tested interoperability locally between nodes before and after this PR. We can upgrade to this for now, and once all nodes are upgraded we will be free to modify those constants as we wish.